Neos.ContentRepository.Core/Classes/NodeType/NodeTypeManager.php

```diff
@@ -198,6 +202,81 @@ public function overrideNodeTypes(array $completeNodeTypeConfiguration): void
         }
     }
 
+    /**
+     * @param NodeType $nodeType
+     * @param NodeName $childNodeName
+     * @throws ChildNodeNotConfigured if the requested childNode is not configured. Check via {@see NodeType::hasAutoCreatedChildNode()}.
+     * @return NodeType
+     */
+    public function getTypeOfAutoCreatedChildNode(NodeType $nodeType, NodeName $childNodeName): NodeType
+    {
+        $childNodeTypeName = $nodeType->getNodeTypeNameOfAutoCreatedChildNode($childNodeName);
+        return $this->getNodeType($childNodeTypeName);
+    }
+
+    /**
+     * Return an array with child nodes which should be automatically created
+     *
+     * @return array<string,NodeType> the key of this array is the name of the child, and the value its NodeType.
+     * @api
+     */
+    public function getAutoCreatedChildNodesFor(NodeType $nodeType): array
```

Comment:
hmm i would like to suggest not to introduce the `.*For$` pattern here ^^ (ending with `For` to shorten it - as i think adding `NodeType` is okay and more readable.)

Reply:
🤷 I find it much more readable like this. `$nodeTypeManager->getAutoCreatedChildNodesFor($nodeType)`